File: 프로그래머스/0/181921. 배열 만들기 2/배열 만들기 2.js
function solution(l, r) {
        let result = [];
        
        for(let i = 1; i < 64; i++){
            let num = parseInt(i).toString(2) * 5;
            if(num >= l && num <= r){
                result.push(num);
            }
        }
        
        return result.length == 0 ? [-1] : result;
}

File: 프로그래머스/0/181921. 배열 만들기 2/배열 만들기 2.java
import java.util.*;

class Solution {
    public int[] solution(int l, int r) {
        List<Integer> result = new ArrayList<>();
        Queue<String> queue = new LinkedList<>();
        
        queue.add("5");
        
        while(!queue.isEmpty()){
            String cur = queue.poll();
            int num = Integer.parseInt(cur);
            
            if (num > r) continue;
            
            if(num >= l && num <= r){
                result.add(num);
            }
            
            if(num * 10 <= r){
                queue.add(cur + "5");
                queue.add(cur + "0");
            }
        }
        
        if (result.isEmpty()) {
            return new int[]{-1};
        }
        
        Collections.sort(result);
        
        return result.stream().mapToInt(Integer::intValue).toArray();
    }
}